### PR TITLE
#313 - Scholarship records view

### PR DESCRIPTION
**Pull request by blms**
## Visual review notes
[Latest Percy build](https://percy.io/3201ecb4/geniza/builds/14329435)

In scope for this PR:
- New background color
    - the overall site background in light mode was previously the exact shade that Gissoo used for the shaded background ("includes" section for scholarship records), so I changed the overall light mode background to `#FFFFFF` to match Gissoo's design
- Changes to spacing of editors in document detail view
- Changes to spacing of tabs in document detail view
- Changes to type sizes in document detail view
- All aspects of scholarship records view (mobile and desktop)
    - I couldn't find a design for the "includes" section for dark mode, so I just used 10% opacity white (as the color Gissoo chose is equivalent to 10% opacity black in light mode)

## What this PR does
- Migrates code from `__str__` into `formatted_display`, and updates `__str__` to just call `strip_tags(self.formatted_display())[:-1]` (the `[:-1]` is because `formatted_display` has a period at the end)
- Updates `formatted_display` to match the Chicago Manual of Style as closely as possible
  - Provides `extra_fields` boolean argument to enable/disable some aspects of the citation
- Implements template markup and style according to Gissoo's designs
- Alters some other styles to match Gissoo's updated designs, such as tabs sizing and spacing
- Refactors the document header (which consists of the document title and the Edit link for admins) into a reusable snippet, since it's used in both this view and the document detail view
  - Changes the `h1` title into a `span` and provides per-page `h1` headings
 
<details>
<summary>Old notes</summary>

Draft PR for #313

## What I've done so far

- Moved code from `__str__` into `formatted_display`, and updated `__str__` to just call `strip_tags(self.formatted_display())[:-1]` (the `[:-1]` is because `formatted_display` has a period at the end)
- Updated `formatted_display` to match the Chicago Manual of Style as closely as possible
  - Note that the proposed format in the issue description differs somewhat from CMS. When I found inconsistencies I erred on the side of accurate CMS as opposed to the format in the issue description.
- Added a migration to add `place_published` and `publisher`
- ~Added a data migration to populate `page_range` from `other_info` (which is where page ranges had been stored previously)~
- Implemented template markup and style according to Gissoo's designs
  - Ran into a couple of issues with the shaded background for the doc relations section (labeled "includes"): 
    - the overall site background in light mode was previously the exact shade that Gissoo used for the shaded background, so I changed the overall light mode background to `#FFFFFF` to match Gissoo's design
    - I couldn't find a design for this for dark mode, so I just used 10% opacity white (as the color Gissoo chose is equivalent to 10% opacity black in light mode)
- Refactored the document header (which consists of the `h1` title and the Edit link for admins) into a reusable snippet, since it's used in both this view and the document detail view
- Altered some other styles to match Gissoo's updated designs, such as tabs sizing and spacing

## Test failures

I still have a test failure that I need some help with:

- `test_old_pgp_edition`
  - The formatted citation adds (n.p., n.d.), per CMS when those pieces of information are missing. I only exclude this altogether when the source type is Unpublished. Since this source type is Book, the n.p. and n.d. still show up. Should I just exclude this if there's not a publisher, location, or date? Or is there another way we can fix this test?

<details><summary>fIxed</summary>

- `MergeIndiaBookSources::test_sources_merged`
  - This one is failing because my data migration to populate page ranges is irreversible. Not sure what to do about that one!
  
 </details>

## Data questions

I have a lot of questions, mostly related to the existing data, so I put them under a details.

<details>
<summary>Questions</summary>

### No publisher, no place, no date

I've been using "n.p." and "n.d." abbreviations for these per CMS. Let me know if they should be omitted instead. I've omitted them already for source type "Unpublished," since those will likely not have publication data, except maybe a date.

### No title

When no title is available, I just use `source_type.type.lower()`, so that links would still work according to Gissoo's designs. It's also somewhat following the CMS guideline to use a descriptive phrase when no title is available. Let me know if that makes sense or if it should just be omitted. (And let me know how I should include links in those cases!)

### Footnote `location` field

Do we want this to be visible in the scholarship records view? Right now I only have it in document detail view under "Editor"—which I'm also wondering, is that just meant to be a list of editors' names?

### Journal issue

Should we add a journal issue field? Issue number is usually included in citations for journal articles.

### Source `edition` field

Should `edition` be an integer instead of a string? That way we can use ordinal numerals (e.g. "5th ed.").

Right now it is only being used once (in the copy of data that I have), to store publication info:

- source id 783
  - `edition="Jerusalem: Shazar"`
  - should get publication place Jerusalem and publisher Shazar
  - also, this entry has incorrect data: its source type should be Book Section, not Article

### Source `other_info` field

Previously, `other_info` was just appended to the end of the citation. Should that remain true? I think it makes more sense to incorporate the "other info" into the data model, so that it can be formatted appropriately per CMS.

#### Publisher data

A lot of the `other_info` fields contain publisher data. I already started by adding a migration with `publication_place` and `publisher` since those are very common. Let me know if I should also populate these in a data migration, or if it makes more sense for admins to just do this in the admin frontend.

- source id 21
  - `other_info="PhD diss. Tel Aviv University"`
  - should get publication place "Tel Aviv", publisher "Tel Aviv University"
  - source type is already "Dissertation"

- source id 628
  - `other_info="Taylor-Schechter Genizah Research Unit, Cambridge University Library"`
  - should get publication place "Cambridge, UK", publisher "Taylor-Schechter Genizah Research Unit, Cambridge University Library"

- source id 675
  - `other_info="Taylor-Schechter Genizah Research Unit, Cambridge University Library"`
  - should get publication place "Cambridge, UK", publisher "Taylor-Schechter Genizah Research Unit, Cambridge University Library"

- source id 723
  - `other_info="Leiden: Brill, 1894"`
  - should get publication place "Leiden", publisher "Brill"
  - year already entered

- source id 770
  - `other_info="PhD Dissertation. The Hebrew University of Jerusalem."`
  - should get publication place "Jerusalem", publisher "The Hebrew University of Jerusalem"
  - source type is already "Dissertation"

- source id 778
  - `other_info="Yaron Ben-Naeh et. al. eds. Jerusalem: Zalman Shazar Center, 2014"`
  - should get publication place "Jerusalem", publisher "Zalman Shazar Center"
  - year already entered

- source id 797
  - `other_info="Wiesbaden: Harrassowitz, 2011, 175–86"`
  - should get publication place "Wiesbaden", publisher "Harrassowitz"
  - page range, year already entered

#### Editors

These contain information about editors. Should we handle this with some kind of `author_type` field or `is_editor` boolean in the `Authorship` model?

- source id 771
  - `other_info="eds. Yvonne Friedman and Joseph Drory."`

- source id 301
  - `other_info=""Hiqre 'Ever va-'Arav : mugashim li-Yehoshu'a Bla'u\r\nEd. H Ben-Shammai"`
  - is the first part of this just the title in Hebrew?

#### Other

Not sure what to do with these.

- source id 17
  - `other_info="unpublished corpus of editions"`
  - source type is already "Unpublished"

- source id 45
  - `other_info="Trans. Goitein in the notes linked below"`
  - if Goitein should be treated as a translator in the citation, that's maybe a good argument for adding an `author_type` field to `Authorship`

- source id 572
  - `other_info="no shelfmark given"`

- source id 767
  - `other_info="no. 1 and 2"`
</details>
</details>

### CODE REVIEW

**Review by rlskoeser — COMMENTED** [review]
This is looking really good — great to see all the progress.

You ask some good questions: I think there's some tension between the previous practice to keep citations fairly minimal and brief and the desire to use CMS. I think you were right to stick to CMS where you saw discrepancies, but we may not want or need full citations.

I'll ask a question on the ticket — let's hold off on the failing test until we get an answer.

When you get the chance, please add a percy snapshot for the scholarship records page.

File: geniza/footnotes/migrations/0014_populate_page_ranges.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+def populate_page_ranges(apps, schema_editor):
+    Source = apps.get_model("footnotes", "Source")
+    sources = Source.objects.exclude(other_info__isnull=True).exclude(
+        other_info__exact=""
+    )
+    for source in sources:
+        reg = re.search(r"(\d+)(–|-)(\d+)", source.other_info)
+        if reg:
+            source.page_range = "%s–%s" % (reg.group(1), reg.group(3))
+            source.save()
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/footnotes/migrations/0014_populate_page_ranges.py:17]: Good point! It should probably all be done manually to catch those cases. Here's the report of all `other_info` fields with any text in them. Should I also put this in the issue for MR/RR to review?

<details>
<summary>other_info</summary>

Source ID 783
Citation: Avraham David, "Between Ashkenaz and the East in the Sixteenth Century: Ashkenazic Jews in the Land of Israel as Reflected in the Cairo Geniza" (in Hebrew), Rishonim ve-Aharonim, From Sages to Savants: Studies Presented to Avraham Grossman (n.p., 2010).
`other_info`: pp. 309–28

Source ID 804
Citation: Geoffrey Khan, "A Judaeo-Arabic Document from Ottoman Egypt in the Rylands Genizah Collection", in From Cairo to Manchester: Studies in the Rylands Genizah Fragments (n.p., 2013).
`other_info`: pp. 233–48

Source ID 799
Citation: Dotan Arad and Esther-Miriam Wagner, "A Letter by Isaac Bayt ʿAṭṭān to Moses b. Judah (1480s)", in A Handbook and Reader of Ottoman Arabic (n.p., 2021).
`other_info`: pp. 143–48

Source ID 797
Citation: Geoffrey Khan, "A Petition to the Fāṭimid Caliph al-Āmir from the Cairo Genizah Concerning an Inheritance", Orientalistische Studien zu Sprache und Literatur Festgabe zum 65. Geburtstag von Werner Diem (n.p., 2011).
`other_info`: Wiesbaden: Harrassowitz, 2011, 175–86

Source ID 789
Citation: Paul Fenton, "A Re-discovered Description of Maimonides by a Contemporary", Maimonidean Studies 5 (n.p., 2008).
`other_info`: 267–91

Source ID 802
Citation: Siam Bhayro, "A Syriac Fragment from the Cairo Genizah", Wiener Zeitschrift für die Kunde des Morgenlandes 94 (n.p., 2004).
`other_info`: pp.39–51

Source ID 628
Citation: Esther-Miriam Wagner, A newly-discovered fragment of a letter written by Maimonides, Fragment of the Month (n.p., 2007).
`other_info`: Taylor-Schechter Genizah Research Unit, Cambridge University Library

Source ID 767
Citation: Andrew Sharf, "An Unknown Messiah of 1096 and the Emperor Alexius", Journal of Jewish Studies 7 (n.p., 1956).
`other_info`: no. 1 and 2

Source ID 572
Citation: Marmorstein, "Beiträge zur Geschichte und Literatur der gaonäischen Periode" (in German), MGWJ 51 (n.p., 1907).
`other_info`: no shelfmark given

Source ID 778
Citation: Dotan Arad, "Cairo's Maghribians and Musta'ribs: From Close Cooperation to Conflict and Estrangement" (in Hebrew), in Studies in Jewish History Presented to Joseph Hacker (n.p., 2014).
`other_info`: Yaron Ben-Naeh et. al. eds. Jerusalem: Zalman Shazar Center, 2014

Source ID 727
Citation: Lévi Israël,  "Document relatif à la «Communauté des fils de Sadoc»" (in French), Revue des études juives 65, n°129 (n.p., 1913).
`other_info`: pp. 24-31

Source ID 723
Citation: Adalbert Merx,  Documents de paléographie hébraïque et arabe (in French) (n.p., 1894).
`other_info`: Leiden: Brill, 1894

Source ID 771
Citation: Dotan Arad, "Documents in Arabic and Judeo-Arabic Relative to the History of the Jewish Community in Jerusalem" (in Hebrew), in The History of Jerusalem: The Mamluk Period (1260-1517) (n.p., 2013).
`other_info`: eds. Yvonne Friedman and Joseph Drory.

Source ID 21
Citation: Ṣabīḥ ʿAodeh, "Eleventh Century Arabic Letters of Jewish Merchants from the Cairo Geniza" (in Hebrew), PhD diss., (n.p., 1992).
`other_info`: PhD diss. Tel Aviv University

Source ID 785
Citation: Jacob Mann, "Glanures de la Gueniza" (in French), Revue des études juives 74, no. 148 (n.p., 1922).
`other_info`: pp. 148–59

Source ID 780
Citation: Alexander Scheiber, "Isaac Ibn Chalfon's Panegyric Poem Addressed to Samuel Han-Nagid: From the Kaufmann Geniza", Acta Orientalia Academiae Scientiarum Hungaricae Vol. 10, No. 1 (n.p., 1960).
`other_info`: pp. 91–95

Source ID 801
Citation: George Kiraz, "Learning Syriac and Garshuni in Early Modern Egypt: Evidence from the Cairo Genizah", Intellectual History of the Islamicate World (n.p., 2020).
`other_info`: pp.1–26

Source ID 17
Citation: S. D. Goitein, Nahray (in Hebrew).
`other_info`: unpublished corpus of editions

Source ID 301
Citation: Mordechai Akiva Friedman, "New Fragments from the Geniza of Maimonides' Responsa (With Addenda to the Published Responsa)" (in Hebrew), in Hebrew and Arabic Studies in Honor of Joshua Blau (n.p., 1993).
`other_info`: Hiqre 'Ever va-'Arav : mugashim li-Yehoshu'a Bla'u
Ed. H Ben-Shammai

Source ID 800
Citation: Friedrich Niessen, "New Testament translations from the Cairo Genizah", Collectanea Christiana Orientalia 6 (n.p., 2009).
`other_info`: pp. 201–22

Source ID 271
Citation: Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991).
`other_info`: 281–345

Source ID 45
Citation: S. Assaf, "Sefer ha-shetarot le-Rav Hai bar Sherira Ga'on" (in Hebrew), Tarbiz 1 (n.p., 1930).
`other_info`: Trans. Goitein in the notes linked below

Source ID 675
Citation: Dotan Arad, Syria’s links with the Jews of Cairo in the 15th and 16th centuries, Fragment of the Month (n.p., 2009).
`other_info`: Taylor-Schechter Genizah Research Unit, Cambridge University Library

Source ID 786
Citation: Dotan Arad, "The Jews of Alexandria in the 15th Century in Light of New Documents" (in Hebrew), Peʿamim 156 (n.p., 2018).
`other_info`: pp. 167–84

Source ID 770
Citation: Dotan Arad, "The Mustaʿrib Jews in Syria, Palestine and Egypt: 1330-1700" (in Hebrew), PhD diss., (n.p., 2013).
`other_info`: PhD Dissertation. The Hebrew University of Jerusalem.

Source ID 788
Citation: Mordechai Akiva Friedman, "The Nagid, the Nasi and the French Rabbis: A Threat to Abraham Maimonides’ Leadership" (in Hebrew), Zion 82 (n.p., 2017).
`other_info`: 193–266

Source ID 781
Citation: "“Let's learn Turkish”: A Turkish-Arabic Glossary from the Cairo Genizah", Turcica 48 (n.p., 2017).
`other_info`: pp. 451–73

</details>

[the remark this answers is not in the document]
Comment (rlskoeser) [reply geniza/footnotes/migrations/0014_populate_page_ranges.py:17]: @blms yes, please go ahead and share it there — they may want to paste it into a google doc or something they can assign a research assistant to go through and work on

File: geniza/footnotes/models.py [hunk 2]
```diff
@@ -144,35 +146,121 @@ def __str__(self):
...
+
+        # italics for book/journal title
+        if self.journal or self.source_type.type == "Dissertation":
+            if self.title and (
+                self.source_type.type in doublequoted_types
+                and not self.languages.count()
+            ):
+                # special case for comma inside doublequotes
+                last_dq = parts[-1].rindex('"')
+                parts[-1] = parts[-1][:last_dq] + "," + parts[-1][last_dq:]
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/footnotes/models.py]: That's right. Both of those cases should be accounted for. If the title is a link, and it is double-quoted, we will still get the comma inside the double quotes (via `rindex`). Here's a few examples handled correctly:
```html
Marina Rustow, <a href="https://docs.google.com/document/d/1sC4IL601LLv76UuM8KhaDfxWfjTlFHnJtz1YeB_1EuI/edit?usp=sharing">"The Fatimid Petition,"</a> <em>Jewish History</em> 32 (n.p., 2019).

Judith Olszowy-Schlanger, <a href="https://www.academia.edu/38307649/Judith_Olszowy-Schlanger_La_lettre_de_divorce_cara%C3%AFte_et_sa_place_dans_les_relations_entre_Cara%C3%AFtes_et_Rabbanites_au_Moyen_Age_Une_%C3%A9tude_de_manuscrits_de_la_Geniza_du_Caire_Revue_des_Etudes_Juives_155_3-4_July-December_1996_261-285">"La lettre de divorce caraïte et sa place dans les relations entre Caraïtes et Rabbanites au Moyen Age (Une étude de manuscrits de la Geniza du Caire)"</a> (in French), <em>Revue des Etudes Juives</em> 155:3–4 (n.p., 1996).
```

Just did some refactoring and commenting on this to hopefully clarify.

File: geniza/corpus/templates/corpus/document_scholarship.html [hunk 3]
```diff
@@ -6,20 +6,28 @@
...
-                        {% if fn.source.authors.exists %}<span class="author">{{ fn.source.all_authors }}</span>{% endif %}
-                        {% if fn.source.year %}<span class="year">{{ fn.source.year }}</span>{% endif %}
-                        <span class="relation">{{ fn.doc_relation }}</span>
-                        {% if fn.location %}<span class="location">{{ fn.location }}</span>{% endif %}
-                        {% if fn.url %}<a href="{{ fn.url }}">{% translate "Link to source" %}</a>{% endif %}
-                    </p></li>
+                    <li class="footnote">
+                        <dt class="sr-only">
+                            {# Translators: accessibility label for a footnote citation in scholarship records view #}
+                            {% translate 'Footnote' %}
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/corpus/templates/corpus/document_scholarship.html]: Ended up changing this `h1` to a `span` and adding an SR-only `h1` to both pages with their own `page_title` context variables.

File: geniza/footnotes/models.py [hunk 4]
```diff
@@ -144,35 +146,121 @@ def __str__(self):
...
+            parts[-1] += "</a>"
+
+        # Add non-English languages as parenthetical
+        if self.languages.count():
+            for lang in self.languages.all():
+                if "English" not in str(lang):
+                    parts.append("(in %s)" % lang)
+
+        # italics for book/journal title
+        if self.journal or self.source_type.type == "Dissertation":
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/footnotes/models.py]: Oops, the comment was out of date! Yes, that should be handled fine. Hopefully, the updated comments should explain what each if statement is checking for.

File: geniza/footnotes/models.py [hunk 5]
```diff
@@ -144,35 +146,121 @@ def __str__(self):
...
+            self.volume and self.source_type.type not in ["Article", "Unpublished"]
+        )
+
+        # Page range
+        if self.page_range:
+            parts[-1] += ","
+            if needs_volume:
+                parts.append("%s:%s" % (self.volume, self.page_range))
+            elif not self.volume:
+                parts.append(self.page_range)
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/footnotes/models.py]: Great question. I found several such examples:

<details>
<summary>Source page range in other_info AND footnote location</summary>

```
Footnote ID: 		 6645
Footnte: 		 Discussion of T-S Ar.30.232 + … (PGPID 20354)
Source ID: 		 783
Source: 		 Avraham David, "Between Ashkenaz and the East in the Sixteenth Century: Ashkenazic Jews in the Land of Israel as Reflected in the Cairo Geniza" (in Hebrew), Rishonim ve-Aharonim, From Sages to Savants: Studies Presented to Avraham Grossman (n.p., 2010), 309–28
Footnote location: 	 314–15
Source other_info: 	 pp. 309–28

Footnote ID: 		 6317
Footnte: 		 Edition and Translation of Moss. I,40 (PGPID 26639)
Source ID: 		 727
Source: 		 Lévi Israël, "Document relatif à la «Communauté des fils de Sadoc»" (in French), Revue des études juives 65, n°129 (n.p., 1913), 24–31
Footnote location: 	 25
Source other_info: 	 pp. 24-31

Footnote ID: 		 6755
Footnte: 		 Edition of CUL Or.1081 2.75.3 (PGPID 31469)
Source ID: 		 801
Source: 		 George Kiraz, "Learning Syriac and Garshuni in Early Modern Egypt: Evidence from the Cairo Genizah," Intellectual History of the Islamicate World (n.p., 2020), 1–26
Footnote location: 	 17–18, 20–22
Source other_info: 	 pp.1–26

Footnote ID: 		 6756
Footnte: 		 Edition of CUL Or.1081 2.75.6 (PGPID 31470)
Source ID: 		 801
Source: 		 George Kiraz, "Learning Syriac and Garshuni in Early Modern Egypt: Evidence from the Cairo Genizah," Intellectual History of the Islamicate World (n.p., 2020), 1–26
Footnote location: 	 3–7
Source other_info: 	 pp.1–26

Footnote ID: 		 6754
Footnte: 		 Edition and Translation of T-S 13J7.8 + … (PGPID 34432)
Source ID: 		 800
Source: 		 Friedrich Niessen, "New Testament translations from the Cairo Genizah," Collectanea Christiana Orientalia 6 (n.p., 2009), 201–22
Footnote location: 	 213–16
Source other_info: 	 pp. 201–22

Footnote ID: 		 6716
Footnte: 		 Edition of T-S NS 311.31 + … (PGPID 9451)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #137a (p.314)
Source other_info: 	 281–345

Footnote ID: 		 6717
Footnte: 		 Edition of DK 123f (PGPID 34267)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #165a (p.315)
Source other_info: 	 281–345

Footnote ID: 		 6718
Footnte: 		 Edition of T-S AS 158.174 (PGPID 22385)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #183a (p.316)
Source other_info: 	 281–345

Footnote ID: 		 6719
Footnte: 		 Edition of ENA 1490.7 (PGPID 12471)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #192a (p.317)
Source other_info: 	 281–345

Footnote ID: 		 6720
Footnte: 		 Edition of JRL Series L 75 (PGPID 34268)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #203a (p.320)
Source other_info: 	 281–345

Footnote ID: 		 6721
Footnte: 		 Edition of T-S AS 147.10 (PGPID 4115)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #262a (p.322)
Source other_info: 	 281–345

Footnote ID: 		 6722
Footnte: 		 Edition of T-S NS 325.81a (PGPID 31252)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #265a (p.323)
Source other_info: 	 281–345

Footnote ID: 		 6715
Footnte: 		 Edition of T-S 10J32.8 + … (PGPID 1553)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #38 (p.311)
Source other_info: 	 281–345

Footnote ID: 		 6714
Footnte: 		 Edition of BL OR 10578H.34 (PGPID 34264)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #3a (p.311)
Source other_info: 	 281–345

Footnote ID: 		 6713
Footnte: 		 Edition of BL OR 5542.30 (PGPID 6224)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #439a (p.334)
Source other_info: 	 281–345

Footnote ID: 		 1688
Footnte: 		 Edition of JRL Gaster heb. ms 1772/3 (PGPID 32340)
Source ID: 		 271
Source: 		 Moshe Gil, "Palestine During the First Muslim Period (634–1099): Additions, Notes, and Corrections" (in Hebrew), Teuda 7 (n.p., 1991), 281–345
Footnote location: 	 #7, p. 333
Source other_info: 	 281–345

Footnote ID: 		 6680
Footnte: 		 Edition of ENA NS I.92 + … (PGPID 11977)
Source ID: 		 788
Source: 		 Mordechai Akiva Friedman, "The Nagid, the Nasi and the French Rabbis: A Threat to Abraham Maimonides’ Leadership" (in Hebrew), Zion 82 (n.p., 2017), 193–266
Footnote location: 	 239–42
Source other_info: 	 193–266
```

</details>

**Comment by blms** [comment]
@rlskoeser I was running into some directional issues with Hebrew appearing in citations; I imagine similar issues would happen with Arabic. As a quick solution I just appended `&lrm;` immediately after every place that Hebrew or Arabic might appear in the citation. Does that seem like a good solution, or do you have an alternative? (Right now it's also appearing in the `__str__` representation of the source, which might not be ideal, but not sure how else to handle for the document detail page)

Without `&lrm;`:

Screen Shot 2021-12-01 at 1 07 25 PM
Screen Shot 2021-12-01 at 1 07 22 PM

With `&lrm;`:

Screen Shot 2021-12-01 at 1 07 35 PM
Screen Shot 2021-12-01 at 1 07 39 PM

**Comment by rlskoeser** [comment]
> @rlskoeser I was running into some directional issues with Hebrew appearing in citations; I imagine similar issues would happen with Arabic. As a quick solution I just appended `&lrm;` immediately after every place that Hebrew or Arabic might appear in the citation. Does that seem like a good solution, or do you have an alternative? (Right now it's also appearing in the `__str__` representation of the source, which might not be ideal, but not sure how else to handle for the document detail page)

Wow, wild. Would you share some examples of this in the db? I remember we set up model translation for some of the source fields but I don't remember which ones; I'm wondering if we know from the database what language this is so we can tag it accordingly for the formatted version. (Although I suspect we did not enable translation for this ...)

If you use the unicode version of the left-to-right and right-to-left override marks instead of the HTML character, will that work for both HTML and text versions?

**Comment by blms** [comment]
> Wow, wild. Would you share some examples of this in the db? I remember we set up model translation for some of the source fields but I don't remember which ones; I'm wondering if we know from the database what language this is so we can tag it accordingly for the formatted version. (Although I suspect we did not enable translation for this ...)

Sure; do you just mean examples like this?

```
Source ID: 		 777
str: 		 	 Mordechai Akiva Friedman, "קטעים חדשים של שאלות ותשובות הרמב"ם‎" (in Hebrew), in מחקרי עדות וגניזה‎ (1981)
title: 			 קטעים חדשים של שאלות ותשובות הרמב"ם
journal: 		 מחקרי עדות וגניזה
languages.all(): 	 Hebrew

Source ID: 		 795
str: 			 Allony, article (in Hebrew), ספונות‎ 6 (1964)
title: 			 None
journal: 		 ספונות
languages.all(): 	 Hebrew
```

FWIW, it seems like they are tagged correctly with language Hebrew. So if we want to just rely on proper tagging, we can include the additional character only when RTL languages are tagged.

> If you use the unicode version of the left-to-right and right-to-left override marks instead of the HTML character, will that work for both HTML and text versions?

Yes, it does! And it works when just printing those string representations now. But, I did notice that it's still failing tests that involve string comparisons.

**Comment by blms** [comment]
This one should be ready for review. There are still a [couple of lingering questions](https://github.com/Princeton-CDH/geniza/issues/313#issuecomment-983907447), but they are very minor. I've provided visual testing notes in the PR description above for @gissoo.

File: geniza/footnotes/models.py [hunk 6]
```diff
@@ -100,6 +101,10 @@ class Source(models.Model):
     page_range = models.CharField(
         max_length=255, blank=True, help_text="Page range for article or book section."
     )
+    publisher = models.CharField(max_length=255, blank=True, help_text="Publisher name")
+    place_published = models.CharField(
```
Comment (rlskoeser) [inline geniza/footnotes/models.py]: please add the new fields to:
- list of source fields that are editable in the admin site (and would you add source url to the list also?) — see https://github.com/Princeton-CDH/geniza/blob/main/geniza/footnotes/admin.py#L115-L124 
- list of csv fields for admin export (see https://github.com/Princeton-CDH/geniza/blob/main/geniza/footnotes/admin.py#L168-L183 )

**Review by rlskoeser — APPROVED** [review]
I think you still need to make the new source fields available in the admin site, but the rest of the changes look good to me.

File: geniza/footnotes/models.py (hunk 6 above)
Comment (blms) [reply geniza/footnotes/models.py]: Forgot about this! 😅 Thanks.

**Pull request merged**